src/crypto/x509.js

```diff
@@ -189,7 +193,36 @@ export default class X509Chain {
 
     getParameterValueByName(parameterName) {
         let signerCert = this.certChain[0];
-        return signerCert[parameterName]?.typesAndValues[0]?.value?.valueBlock?.value || null;
+
+        return signerCert[parameterName].typesAndValues|| [];
+    }
+
+    getSubjectValue() {
+        const subjectValues = this.getParameterValueByName('subject');
+        const subjectName = subjectValues.find((v) => v.type === OIDs.commonName);
```

Comment:
Let's use the same fallback logic as for the issuer here, since the structure of these fields is identical.
In case common name is missing, we could use organization name.